example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt

```diff
@@ -145,6 +176,71 @@ class TruidSignupFlow(
         }
     }
 
+    @GetMapping(
+        path = ["/truid/v1/presentation"],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    suspend fun getPresentation(): PresentationResponse {
+        val accessToken = getAccessToken()
+
+        val getPresentationUri = URIBuilder(truidPresentationEndpoint)
+            .addParameter("claims", "truid.app/claim/email/v1")
+            .build()
+
+        return webClient
+            .get()
+            .uri(getPresentationUri)
+            .accept(APPLICATION_JSON)
+            .header(AUTHORIZATION, "Bearer $accessToken")
+            .retrieve()
+            .awaitBody()
+    }
+
+    private suspend fun getAccessToken(): String {
+        val (tokenResponse, expires) = getPersistedToken() ?: throw Forbidden(
+            "access_denied",
+            "No access_token or refresh_token found"
+        )
+
+        return if (expires > LocalDateTime.now()) {
+            tokenResponse.accessToken
+        } else {
+            //If access token is expired, use refresh token to get a new one
+            refreshToken()
+            getPersistedToken()!!.first.accessToken
+        }
+    }
+
+    private suspend fun refreshToken() {
+        // Check if token was refreshed by another thread, two refreshes with same refresh token
```

Comment:
Not true in our case, but might be good to point out as we might want it in the future?